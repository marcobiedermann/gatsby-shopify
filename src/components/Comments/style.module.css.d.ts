declare const styles: {
  readonly comments: string;
};
export = styles;
