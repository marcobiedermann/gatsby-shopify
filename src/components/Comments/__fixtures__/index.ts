import { CommentsProps } from '..';
import comment from '../../Comment/__fixtures__';

const comments: CommentsProps = {
  comments: [comment],
};

export default comments;
