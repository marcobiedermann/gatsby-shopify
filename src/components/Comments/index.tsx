import React, { FC } from 'react';
import Comment, { CommentProps } from '../Comment';
import styles from './style.module.css';

export interface CommentsProps {
  comments: CommentProps[];
}

const Comments: FC<CommentsProps> = (props) => {
  const { comments } = props;

  return (
    <ol className={styles.comments}>
      {comments.map((comment) => (
        <li key={comment.shopifyId}>
          <Comment {...comment} />
        </li>
      ))}
    </ol>
  );
};

export default Comments;
