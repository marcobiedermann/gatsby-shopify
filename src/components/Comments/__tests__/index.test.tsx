import { render } from '@testing-library/react';
import React from 'react';
import Comments from '..';
import comments from '../__fixtures__';

describe('comments component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Comments {...comments} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
