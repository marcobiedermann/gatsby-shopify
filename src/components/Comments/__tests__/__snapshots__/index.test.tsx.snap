// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments component should render correctly 1`] = `
<DocumentFragment>
  <ol
    class="comments"
  >
    <li>
      <div>
        <div />
        <div>
          <span>
            John Doe
          </span>
        </div>
      </div>
    </li>
  </ol>
</DocumentFragment>
`;
